hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -1191,7 +1191,20 @@ def zero_examples(self, ex):
                 self.buffer[:u] + hbytes(used) + self.buffer[v:]
             )
 
-    def minimize_duplicated_blocks(self):
+    def duplicated_block_suffixes(self):
+        duplicates = defaultdict(list)
+        for b in self.blocks:
+            duplicates[non_zero_suffix(self.buffer[b.start : b.end])].append(b.index)
+        return sorted(
+            [
+                (b, indices)
```

Comment:
It's slightly confusing that this `b` has a different type from the `b` a few lines above, and then ends up becoming `block` in the function argument below.

Might be clearer to expand this to `block_bytes` here and below.